pybop/optimisers/scipy_optimisers.py

```diff
@@ -65,17 +65,19 @@ def _run(self):
 
         Returns
         -------
-        x : numpy.ndarray
-            The best parameter set found by the optimization.
-        final_cost : float
-            The final cost associated with the best parameters.
+        result : pybop.Result
+            The result of the optimisation including the optimised parameter values and cost.
         """
-        self.result = self._run_optimiser()
+        result = self._run_optimiser()
 
-        self.result.final_cost = self.cost(self.result.x)
-        self._iterations = self.result.nit
+        self._iterations = result.nit
```

Comment:
I see that we set this attr, but don't use it the results class below. Is it needed for something else or do you think we can remove it? Alternatively, we could pass it directly to `Results` if that helps align the API between SciPy and Pints?

Reply:
This property is currently used in the tests, but we can switch to using the property of the result.